server/odc-service/src/main/java/com/oceanbase/odc/service/flow/task/MultipleDatabaseChangeRuntimeFlowableTask.java

```diff
@@ -0,0 +1,232 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.oceanbase.odc.service.flow.task;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.flowable.engine.delegate.DelegateExecution;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import com.oceanbase.odc.common.json.JsonUtils;
+import com.oceanbase.odc.common.trace.TaskContextHolder;
+import com.oceanbase.odc.common.trace.TraceContextHolder;
+import com.oceanbase.odc.core.shared.constant.FlowStatus;
+import com.oceanbase.odc.core.shared.constant.TaskType;
+import com.oceanbase.odc.metadb.flow.FlowInstanceEntity;
+import com.oceanbase.odc.metadb.flow.ServiceTaskInstanceEntity;
+import com.oceanbase.odc.metadb.flow.ServiceTaskInstanceRepository;
+import com.oceanbase.odc.metadb.task.TaskEntity;
+import com.oceanbase.odc.service.flow.FlowInstanceService;
+import com.oceanbase.odc.service.flow.FlowableAdaptor;
+import com.oceanbase.odc.service.flow.model.CreateFlowInstanceReq;
+import com.oceanbase.odc.service.flow.model.FlowInstanceDetailResp;
+import com.oceanbase.odc.service.flow.model.FlowTaskExecutionStrategy;
+import com.oceanbase.odc.service.flow.task.model.MultipleDatabaseChangeParameters;
+import com.oceanbase.odc.service.flow.task.model.MultipleDatabaseChangeTaskResult;
+import com.oceanbase.odc.service.flow.util.FlowTaskUtil;
+import com.oceanbase.odc.service.iam.auth.AuthenticationFacade;
+import com.oceanbase.odc.service.task.TaskService;
+
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * @author: zijia.cj
+ * @date: 2024/3/29
+ */
+@Slf4j
+public class MultipleDatabaseChangeRuntimeFlowableTask extends BaseODCFlowTaskDelegate<Void> {
+
+    private volatile boolean isSuccessful = false;
+    private volatile boolean isFailure = false;
+
+    private Integer batchId;
+
+    private Integer batchNumber;
+
+    @Autowired
+    private FlowInstanceService flowInstanceService;
+
+    @Autowired
+    private FlowableAdaptor flowableAdaptor;
+
+    @Autowired
+    private ServiceTaskInstanceRepository serviceTaskInstanceRepository;
+    @Autowired
+    private AuthenticationFacade authenticationFacade;
+
+    @Override
+    public boolean cancel(boolean mayInterruptIfRunning, Long taskId, TaskService taskService) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public boolean isCancelled() {
+        return false;
+    }
+
+    @Override
+    protected Void start(Long taskId, TaskService taskService, DelegateExecution execution)
```

Comment:
the multiple database change task will pending until all sub task completed ?
seems not a good solution, what if current task interrupted due odc node crash ?

Reply:
Maintain the current practice and let the subsequent task framework handle odc node crashes  scenario